x-pack/plugins/asset_inventory/common/types_api.ts

```diff
@@ -8,19 +8,48 @@
 export type AssetKind = 'unknown' | 'node';
 export type AssetType = 'k8s.pod' | 'k8s.cluster' | 'k8s.node';
 
-export interface Asset {
+export interface ECSDocument {
   '@timestamp': string;
+  kubernetes?: EcsKubernetesFieldset;
+  orchestrator?: EcsOrchestratorFieldset;
+  cloud?: EcsCloudFieldset;
+}
+
+export type AssetStatus = 'CREATING' | 'ACTIVE' | 'DELETING' | 'FAILED' | 'UPDATING' | 'PENDING';
```

Comment:
Is this specific to k8s cluster or you expect this to be a generic field?

Reply:
Not 100% sure yet, I imagine most cloud assets will have something similar to this. I'm not sure if the values are going to align enough for us to use a single set of valid values. Already with Azure, it got fairly weird/complicated (they have provisioningState and powerState values that both _sort of_ align with AWS status)